common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsAuthorizationRequest.java

```diff
@@ -22,7 +22,325 @@
+     */
+    private String mLoginHint;
+    private String mUid;
+    private String mUtid;
+    private String mDisplayableId;
+    /**
+     *  Used to secure authorization code grants via Proof Key for Code Exchange (PKCE) from a native client.
+     */
+    private PKCEChallenge mPKCEChallenge;
+    private String mExtraQueryParam;
+    private String mSliceParameters;
+    private String mLibraryVersion;
+
+    public UUID getCorrelationId() {
+        return mCorrelationId;
+    }
+
+    public void setCorrelationId(final UUID correlationId) {
+        mCorrelationId = correlationId;
+    }
+
+    public String getAuthorizationEndpoint() {
+        return mAuthorizationEndpoint;
+    }
+
+    public void setAuthorizationEndpoint(final String authorizationEndpoint) {
+        mAuthorizationEndpoint = authorizationEndpoint;
+    }
+
+    public Set<String> getExtraScopesToConsent() {
+        return mExtraScopesToConsent;
+    }
+
+    public String getLoginHint() {
+        return mLoginHint;
+    }
+
+    public void setLoginHint(final String loginHint) {
+        mLoginHint = loginHint;
+    }
+
+    public String getUid() {
+        return mUid;
+    }
+
+    public void setUid(final String uid) {
+        mUid = uid;
+    }
+
+    public String getUtid() {
+        return mUtid;
+    }
+
+    public void setUtid(final String utid) {
+        mUtid = utid;
+    }
+
+    public String getDisplayableId() {
+        return mDisplayableId;
+    }
+
+    public void setDisplayableId(final String displayableId) {
+        mDisplayableId = displayableId;
+    }
+
+    public MicrosoftStsPromptBehavior getPromptBehavior() {
+        return mPromptBehavior;
+    }
+
+    public void setPromptBehavior(final MicrosoftStsPromptBehavior promptBehavior) {
+        mPromptBehavior = promptBehavior;
+    }
+
+    public PKCEChallenge getPKCEChallenge() {
+        return mPKCEChallenge;
+    }
+
+    public void setPKCEChallenge(final PKCEChallenge pKCEChallenge) {
+        mPKCEChallenge = pKCEChallenge;
+    }
+
+    public String getExtraQueryParam() {
+        return mExtraQueryParam;
+    }
+
+    public void setExtraQueryParam(final String extraQueryParam) {
+        mExtraQueryParam = extraQueryParam;
+    }
+
+    public String getSliceParameters() {
+        return mSliceParameters;
+    }
+
+    public void setmSliceParameters(final String sliceParameters) {
+        mSliceParameters = sliceParameters;
+    }
+
+    public URL getAuthority() {
+        return mAuthority;
+    }
+
+    public void setAuthority(final URL authority) {
+        mAuthority = authority;
+    }
+
+    /**
+     * @return String of MSAL version name.
+     */
+    public String getLibraryVersion() {
+        return mLibraryVersion;
+    }
+
+    /**
+     * The calling MSAL library should pass in its version name.
+     *
+     * @param libraryVersion String of the MSAL library version name.
+     */
+    public void setLibraryVersion(final String libraryVersion) {
```

Comment:
Why do we need this?

Reply:
If this telemetry or something we really need to evaluate why this would appropriate -- common is its own thing and should be air-gapped from ADAL/MSAL as much as logically possible